ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -623,27 +630,46 @@ private static void maybeCreateProcessingLogStream(
       final KsqlEngine ksqlEngine,
       final CommandQueue commandQueue
   ) {
+    final Set<String> sourceNames = ksqlEngine.getMetaStore().getAllDataSources()
+        .keySet().stream().map(SourceName::name)
+        .collect(Collectors.toSet());
+
     if (!config.getBoolean(ProcessingLogConfig.STREAM_AUTO_CREATE)
-        || !commandQueue.isEmpty()) {
+        || sourceNames.contains(config.getString(ProcessingLogConfig.STREAM_NAME))) {
       return;
     }
 
-    final PreparedStatement<?> statement = ProcessingLogServerUtils
-        .processingLogStreamCreateStatement(config, ksqlConfig);
-    final Supplier<ConfiguredStatement<?>> configured = () -> ConfiguredStatement.of(
-        statement, Collections.emptyMap(), ksqlConfig);
-
+    final TransactionalProducer transactionalProducer =
+        commandQueue.createTransactionalProducer();
     try {
+      transactionalProducer.initTransactions();
+      transactionalProducer.beginTransaction();
+  
```

Comment:
We might wind up writing multiple CS messages for the log stream now, which is never supposed to happen. When we fix up the command runner to no longer skip commands, this will result in it getting stuck on the repeats. I think it would be better for this PR to keep the old behavior of checking the end offset (after starting the txn), and accept that if we have to abort we won't create the log stream. When you refactor, this should get fixed anyway.

Reply:
I'll revert and track this fix here. https://github.com/confluentinc/ksql/issues/3768